File: packages/react-native-messaging-core/src/types/ProviderInConversation.ts
import { Provider } from './Provider';

export class ProviderInConversation {
  provider: Provider;
  typingAt?: Date;
  seenUntil?: Date;

  constructor(provider: Provider, typingAt?: Date, seenUntil?: Date) {
    this.provider = provider;
    this.typingAt = typingAt;
    this.seenUntil = seenUntil;
  }
}

File: packages/react-native-messaging-core/ios/Models/ProviderInConversation.swift
import Foundation
import NablaMessagingCore

extension ProviderInConversation {
    var dictionaryRepresentation: [String: Any] {
        var result = [String: Any]()
        result["provider"] = provider.dictionaryRepresentation
        result["typingAt"] = typingAt?.iso8601String
        result["seenUntil"] = seenUntil?.iso8601String
        return result
    }
}
